views/components/hero.njk

```diff
@@ -1,18 +1,28 @@
 {% import "./headers.njk" as headers %}
 
-{% macro superHero(name, title, content, caption = false) %}
-    <div class="hero hero--major hero--super {% if name %}hero--{{ name }}{% endif %}">
-        <section class="hero__inner">
-            <div class="hero__content inner">
-                <div class="hero__header" id="content">
-                    <h1 class="hero__title">{{ title | safe }}</h1>
-                    {{ content | safe }}
-                </div>
-                {% if caption %}
-                    <span class="hero__caption u-caption u-desktop-only">{{ caption }}</span>
-                {% endif %}
+{% macro superHero(title, content, imageDefault, imageCandidates) %}
+    <div class="super-hero js-random-hero" data-image-candidates='{{ imageCandidates | jsonStringify | safe }}'>
```

Comment:
Do you think it'd be neater to output a `<script>` block here (with a unique ID somehow) and refer to it in the `data` attribute, and then just grab it directly in the JS, rather than parsing JSON? Or would that just increase the chance of errors with global vars etc?

Reply:
I personally like the self-container nature of this, even with the extra step. Prefer avoid global variables if possible.

Reply:
🇹🇭

Reply:
👍